lib/keyring_liberator.rb

```diff
@@ -24,9 +24,25 @@ def self.get_keyring_named(name)
       get_all_keyrings.find { |k| k.name == name }
     end
 
+    def self.get_current_keyring(name, cwd)
+      found_by_name = name && get_all_keyrings.find { |k| k.name == name && k.path == cwd.to_s }
+      found_by_name || KeyringLiberator.get_keyring(cwd)
+    end
+
+    def self.get_all_keyrings_named(name)
+      get_all_keyrings.find_all { |k| k.name == name }
+    end
+
     def self.save_keyring(keyring)
       keys_dir.mkpath
 
+      if get_keyring_named(keyring)
+        ui = Pod::UserInterface
+        ui.puts "About to create a duplicate keyring file for project #{keyring.name.green}"
+        ui.puts "\nPress enter to continue, or `ctrl + c` to cancel"
+        ui.gets
```

Comment:
I haven't tested this out yet, it should run when running `pod install` for the first time in a duplicated project.